code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -0,0 +1,303 @@
+	return TRUE
+
+/datum/hive_status/proc/remove_xeno(var/mob/living/carbon/Xenomorph/X)
+	// Remove() returns 1 if it removes an element from a list
+	if(!xenos_by_tier[X.tier].Remove(X))
+		stack_trace("failed to remove a xeno from hive status tier list, nothing was removed!?")
+		return FALSE
+
+	if(!xenos_by_typepath[X.caste_base_type])
+		stack_trace("trying to remove an invalid typepath from hivestatus list")
+		return FALSE
+
+	if(!xenos_by_typepath[X.caste_base_type].Remove(X))
+		stack_trace("failed to remove a xeno from hive status typepath list, nothing was removed!?")
+		return FALSE
+
+	post_removal(X)
+	return TRUE
+
+/datum/hive_status/proc/unbury_all_larva()
+	return
+
+/datum/hive_status/proc/start_queen_timer()
+	return
+
+/datum/hive_status/normal
+
+/datum/hive_status/normal/on_queen_death(var/mob/living/carbon/Xenomorph/Queen/Q)
+	if(living_xeno_queen != Q)
+		return
+
+	if(!SSticker.mode.stored_larva)
+		return
+
+	SSticker.mode.stored_larva = round(SSticker.mode.stored_larva * ((Q.upgrade+1)/6.0)) // 83/66/50/33 for ancient/elder emp/elder queen/queen
+
+	INVOKE_ASYNC(src, .proc/unbury_all_larva) // this is potentially a lot of calls so do it async
+
+/datum/hive_status/normal/unbury_all_larva()
+	var/turf/larva_spawn
+	while(SSticker.mode.stored_larva > 0) // still some left
+		larva_spawn = pick(GLOB.xeno_spawn)
+		new /mob/living/carbon/Xenomorph/Larva(larva_spawn)
+		SSticker.mode.stored_larva--
+		CHECK_TICK // lets not lag everything
+
+	start_queen_timer() // spawning is done now start it
+
+/datum/hive_status/normal/start_queen_timer()
+	if(!SSticker?.mode)
+		return
+	if(length(xenos_by_typepath[/mob/living/carbon/Xenomorph/Larva]) || length(xenos_by_typepath[/mob/living/carbon/Xenomorph/Drone]))
+		SSticker.mode.queen_death_countdown = world.time + QUEEN_DEATH_COUNTDOWN
+		addtimer(CALLBACK(SSticker.mode, /datum/game_mode.proc/check_queen_status, queen_time), QUEEN_DEATH_COUNTDOWN)
+	else
+		SSticker.mode.queen_death_countdown = world.time + QUEEN_DEATH_NOLARVA
+		addtimer(CALLBACK(SSticker.mode, /datum/game_mode.proc/check_queen_status, queen_time), QUEEN_DEATH_NOLARVA)
+
+/datum/hive_status/corrupted
+	name = "Corrupted"
+	hivenumber = XENO_HIVE_CORRUPTED
+	prefix = "Corrupted "
+	color = "#00ff80"
+
+/datum/hive_status/corrupted/post_add(var/mob/living/carbon/Xenomorph/X)
+	. = ..()
+	X.add_language("English")
+
+/datum/hive_status/corrupted/post_removal(var/mob/living/carbon/Xenomorph/X)
+	. = ..()
+	X.remove_language("English")
+
+/datum/hive_status/alpha
+	name = "Alpha"
+	hivenumber = XENO_HIVE_ALPHA
+	prefix = "Alpha "
+	color = "#cccc00"
+
+/datum/hive_status/beta
+	name = "Beta"
+	hivenumber = XENO_HIVE_BETA
+	prefix = "Beta "
+	color = "#9999ff"
+
+/datum/hive_status/zeta
+	name = "Zeta"
+	hivenumber = XENO_HIVE_ZETA
+	prefix = "Zeta "
+	color = "#606060"
+
+/mob/living/carbon/Xenomorph/proc/set_initial_hivenumber()
+	add_to_hive_by_hivenumber(hivenumber, force=TRUE)
+
+	if(isxenolarva(src))

```

Comment:
This can go in `/mob/living/carbon/Xenomorph/Larva/set_initial_hivenumber()`

Reply:
I actually unfucked larva and now they use generate_name